apps/api/src/purchase-orders/purchase-orders.service.ts

```diff
@@ -0,0 +1,90 @@
+import { Injectable, NotFoundException, Logger } from '@nestjs/common';
+import { PurchaseOrders } from "@prisma/client";
+import { CreatePurchaseOrderDto } from './dto/create-purchase-order.dto';
+import { UpdatePurchaseOrderDto } from './dto/update-purchase-order.dto';
+import { PrismaService } from "../prisma.service";
+
+@Injectable()
+export class PurchaseOrdersService {
+  constructor(private prisma: PrismaService) {
+  }
+
+  // Just a note that the naming convention here seems inconsistent. In prisma, PurchaseOrders is plural while ParentItem is singular
+  async findAll(sortField: string, sortOrder: string): Promise<PurchaseOrders[]> {
+
+    const purchaseOrders = await this.prisma.purchaseOrders.findMany({
+      /* 
+        There is a problem here that should be noted between Prisma and SQLite specifically for dates. Prisma maps DateTime fields to a numeric in SQLite. 
+        The seed data stored the dates in ISO format but when creating or update records, it will be stored as an integer which will make it seem like the sort is broken. 
+        For more info, see: https://github.com/prisma/prisma/issues/8510
+      */
+      orderBy: {
+        [sortField]: sortOrder,
+      },
+      include: {
+        purchase_order_line_items: true,
+      },
+    });
+
+    return purchaseOrders.map(order => ({
+      ...order,
+      total_quantity: order.purchase_order_line_items.reduce((sum, item) => sum + item.quantity, 0),
+      total_cost: order.purchase_order_line_items.reduce((sum, item) => sum + item.quantity * Number(item.unit_cost), 0),
+    }));
```

Comment:
The Date Handling bullet point is addressed above. In terms of the second bullet point around data transformation, since Prisma doesn't support complex aggregation directly in its API, I'm willing to leave the code as is rather than implementing raw SQL in the code. If the dataset size increases to where it is no longer efficient, then we can refactor